akka-edge-docs/src/main/paradox/guide/1-local-drone-control-service.md

```diff
@@ -0,0 +1,238 @@
+
+### Journal storage
+
+In this sample we use Akka Persistence R2DBC with the H2 in-process database, with a file backe storage. H2 requires no
+additional external database service so can be convenient for both development and production usage where only a single
+node interacts with the journal and overhead needs to be kept low. 
+
+It is of course also possible to instead use a separate standalone database such as for example PostgreSQL.
+
+Config to use H2 looks like this:
+
+Scala
+:  @@snip [persistence.conf](/samples/grpc/local-drone-control-scala/src/main/resources/persistence.conf) { }
+
+Java
+:  @@snip [persistence.conf](/samples/grpc/local-drone-control-java/src/main/resources/persistence.conf) { }
+
+In addition to the configuration, the following additional dependencies are needed in the project build:
+
+@@dependency [sbt,Maven,Gradle] {
+group=com.h2database
+artifact=h2
+version=$h2.version$
+group2=io.r2dbc
+artifact2=r2dbc-h2
+version2=$r2dbc-h2.version$
+}
+
+### gRPC Service API for the drone communication
+
+To allow drones to actually use the service we need a public API reachable over the network. For this we will use @extref[Akka gRPC](akka-grpc:)
+giving us a type safe, efficient protocol that allows clients to be written in many languages.
+
+The service descriptor for the API is defined in protobuf, it implements the report command that entity accepts but not
+one matching the get location command:
+
+Scala
+:  @@snip [local.drones.drone_api.proto](/samples/grpc/local-drone-control-scala/src/main/protobuf/local/drones/drone_api.proto) { }
+
+Java
+:  @@snip [local.drones.drone_api.proto](/samples/grpc/local-drone-control-java/src/main/protobuf/local/drones/drone_api.proto) { }
+
+When compiling the project the Akka gRPC @scala[sbt]@java[maven] plugin generates a service interface for us to implement.
+Our implementation of it interacts with the entity:
+
+Scala
+:  @@snip [DroneServiceImpl.scala](/samples/grpc/local-drone-control-scala/src/main/scala/local/drones/DroneServiceImpl.scala) { }
+
+Java
+:  @@snip [DroneServiceImpl.java](/samples/grpc/local-drone-control-java/src/main/java/local/drones/DroneServiceImpl.java) { }
+
+Finally, we need to start the HTTP server, making service implementation available for calls from drones:
+
+Scala
+:  @@snip [LocalDroneControlServer.scala](/samples/grpc/local-drone-control-scala/src/main/scala/local/drones/LocalDroneControlServer.scala) { #bind }
+
+Java
+:  @@snip [LocalDroneControlServer.java](/samples/grpc/local-drone-control-java/src/main/java/local/drones/LocalDroneControlServer.java) { #bind }
+
+The Akka HTTP server must be running with HTTP/2 to serve gRPC, this is done through config:
+
+Scala
+:  @@snip [grpc.conf](/samples/grpc/local-drone-control-scala/src/main/resources/grpc.conf) { #http2 }
+
+Java
+:  @@snip [grpc.conf](/samples/grpc/local-drone-control-java/src/main/resources/grpc.conf) { #http2 }
+
```

Comment:
`Main` is missing?

Reply:
I pretty much mimic:ed the first step of the distributed cluster guide here, where we don't show main, but I'll add it. 👍